### PR TITLE
br: add pitr restore

### PR DESCRIPTION
**Pull request by WizardXiao**
### What problem does this PR solve?

Ref https://github.com/pingcap/tidb-operator/issues/4645
### What is changed and how does it work?

**The operation of pitr restore.**
1. set `restoreMode` to be `pitr`
2. set `pitrRestoredTs`
3. set log backup storage info to original spec's storage provider
4. set full backup into `pitrFullBackupStorageProvider`

**The technical implementation details.**
pitr restore will start a job and run br command about pitr restore, like `./br restore point --storage=xxx --full-backup-storage=xxx  --restored-ts=xxx`
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
add br pitr restore.
```

### CODE REVIEW

File: cmd/backup-manager/app/cmd/backup.go [hunk 1]
```diff
@@ -46,6 +47,10 @@ func NewBackupCommand() *cobra.Command {
 	cmd.Flags().BoolVar(&bo.TLSClient, "client-tls", false, "Whether client tls is enabled")
 	cmd.Flags().BoolVar(&bo.TLSCluster, "cluster-tls", false, "Whether cluster tls is enabled")
 	cmd.Flags().BoolVar(&bo.SkipClientCA, "skipClientCA", false, "Whether to skip tidb server's certificates validation")
+	cmd.Flags().StringVar(&bo.Mode, "mode", string(v1alpha1.BackupModeSnapshot), "backup mode, which is log or snapshot(default)")
+	cmd.Flags().StringVar(&bo.SubCommand, "subcommand", string(v1alpha1.LogStartCommand), "the log backup subcommand")
```
Comment (grovecai) [inline cmd/backup-manager/app/cmd/backup.go:51]: flags usage is not accurate
"backup mode, which is log or snapshot(default)" => "backup mode, which is pitr or snapshot(default)"

**Comment by grovecai** [comment]
rest LGTM

File: cmd/backup-manager/app/cmd/backup.go (hunk 1 above)
Comment (WizardXiao) [reply cmd/backup-manager/app/cmd/backup.go:51]: ok, i think you mean the restore cmd is not accurate in [here](https://github.com/pingcap/tidb-operator/pull/4694/files#diff-acd3036d4ac2e01282ab2bf1035fa449ecaeb9abc5ff8f5583b10e63aabc0d52R52)

Comment (grovecai) [reply cmd/backup-manager/app/cmd/backup.go:51]: yes, you are right

**Review by grovecai — APPROVED** [review]
LGTM

**Review by WangLe1321 — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]

**Review by KanShiori — APPROVED** [review]

**Pull request merged**